senator/game.py

```diff
@@ -78,45 +97,93 @@ def nash_conv(self) -> float:
         nash_convexity = 0
         for player in range(self._num_players):
             instantaneous_regret = self._get_instaneous_regret(player, strategy)
-            nash_convexity += np.sum(instantaneous_regret * strategy[player])
+            assert np.max(instantaneous_regret) >= 0
+            nash_convexity += np.max(instantaneous_regret)
 
         return nash_convexity
 
 
 class Game:
-    def __init__(self, initial_utility: np.ndarray, votes: np.ndarray):
-        assert len(initial_utility.shape) == 2, "Initial utility should be a 2D array"
+    def __init__(
+        self,
+        initial_utility: np.ndarray,
+        votes: np.ndarray,
+        is_owner: np.ndarray,
+        owner_loss: float = 1,
+        other_loss: float = 0,
+        owner_trashold: float = 0,
+    ) -> None:
+
+        self.owner_loss = owner_loss
+        self.other_loss = other_loss
+        self.owner_trashold = owner_trashold
+        self.is_owner = is_owner
         self.votes = votes
         self.initial_utility = initial_utility
-        root_utility_matrix = get_utility_matrix(initial_utility, votes)  # TODO: see what it takes to get the utility matrix
+        root_utility_matrix = get_utility_matrix(initial_utility, votes, is_owner)
         self.root = Node(parent=None, utility_matrix=root_utility_matrix)
 
-    def add_child(self, node: Node, joint_action: np.ndarray, current_utility: np.ndarray, votes: np.ndarray) -> None:
-        child_utility_matrix = get_utility_matrix(current_utility, votes)  # TODO: see what it takes to get the utility matrix
-        node.children[tuple(joint_action)] = Node(parent=node, utility_matrix=child_utility_matrix)
-
-    def solve_node_via_rm_plus(self, node: Node, iterations: int = 1000) -> None:
+    def add_child(
+        self,
+        node: Node,
+        joint_action: np.ndarray,
+        current_utility: np.ndarray,
+        votes: np.ndarray,
+    ) -> None:
+        child_utility_matrix = get_utility_matrix(
+            current_utility,
+            votes,
+            self.is_owner,
+            self.owner_loss,
+            self.other_loss,
+            self.owner_trashold,
+        )  # TODO: see what it takes to get the utility matrix
+        node.children[tuple(joint_action)] = Node(
```

Comment:
Před tenhle řádek bych dal 
```python
assert tuple(joint_action) not in node.children.keys()
```
protože předpokládám že nechceme přidávat node child když už nějaký existuje. Nebo? Co ty na to @DavidSych ?

Reply:
Fakt tady chceme `assert`? Nevim jak tohle navazuje na zbytek kodu, ale určitě nechceme `Node` premazat. Takže možná jen 
```
if tuple(joint_action) not in node.children.keys():
     node.children[tuple(joint_action)] = Node(...)
```

Reply:
Já myslím že jakmile voláme `add_child` na něco co už ten child má tak je to docela fuckup ne?

Reply:
Ale je to synkův kód, tak ať se s tím děje vůle ne boží, ale jeho

Reply:
Asi souhlas no :D Já si představoval, že ty děti se budou přidávat pro náhodně vygenerované akce z equilibiálních strategií. Takže se může stát, že nasampluješ tu joint akci 2x, ale můžeš ten if hodit i mimo tuhle fci.

Reply:
Jasný, jako může být že ten if bude tady, asi to dává smysl. Ale pak bych tenhle fakt hodil do docstringu té metody.